### PR TITLE
Disable some actions while no database is open

### PR DESCRIPTION
**Pull request by motokito**
I am working on this issue https://github.com/JabRef/jabref/issues/463

I have disabled and greyed out all menu items and toolbar buttons that should not be active if no database is open. (take a look at screenshots)

2016-09-05_14h06_41
2016-09-05_14h08_15
2016-09-05_14h08_31
2016-09-05_14h08_52
2016-09-05_14h09_09
2016-09-05_14h09_22
2016-09-05_14h11_28
2016-09-05_14h11_41
2016-09-05_14h11_55
2016-09-05_14h12_04
2016-09-05_14h12_20

- [x] Change in CHANGELOG.md described
- [x] Insert screenshots of menu item and toolbar item, if no database is open.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
I tested your changes and the Rank is disabled, meaning, I can not click on it, but the Icon is not greyed out. 
However, I just printed out all items in `createDisabledIconsForMenuEntries`  and there is no `Rank`, so for one reason I don't know it is not in the list. 
```
  JMenuItem item = (JMenuItem) subElement;
                System.out.println(item.getText());
            if (item.getIcon() instanceof IconTheme.FontBasedIcon) {
``
Furthermore, I printed out all actions and Component in the method
`private static void setEnabled(List<Object> list, boolean enabled) {`
And there is the rank as component listed, so the whole component is set to disabled. 

My best guess, would be that the disabled icon for the special fields is not correctly set, because the pushtoExternal and the rank are both custom JButtons, while the rest are MenuItems.
Mabye sth with the SpecialField.

I can not say it definitely

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java [hunk 1]
```diff
@@ -1543,6 +1551,12 @@ private static void setEnabled(List<Object> list, boolean enabled) {
             }
             if (o instanceof Component) {
                 ((Component) o).setEnabled(enabled);
+                if (o instanceof JPanel) {
+                    JPanel root = (JPanel) o;
+                    for (int index = 0; index < root.getComponentCount(); index++) {
+                        root.getComponent(index).setEnabled(enabled);
+                    }
+                }
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: rename variable o please.

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java [hunk 2]
```diff
@@ -1689,6 +1703,20 @@ public void createDisabledIconsForMenuEntries(MenuElement menuElement) {
         }
     }
 
+    public void createDisabledIconsForButtons(Container container) {
+        for (int index = 0; index < container.getComponentCount(); index++) {
+            Component component = container.getComponent(index);
+            if (component instanceof JButton) {
+                JButton item = (JButton) component;
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: Please rename `item` to `button`

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java [hunk 3]
```diff
@@ -1471,6 +1476,7 @@ private void createToolBar() {
                 specialFieldButtons.add(button);
             }
         }
+
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: Why did you add this empty line?

**Comment by boceckts** [comment]
Only minor comments regarding code refactoring from my side. Good job :+1:

**Comment by stefan-kolb** [comment]
Can you add some screenshots what has changed please?

**Comment by Siedlerchr** [comment]
Ah, cool, you found a way to disable those buttons menu things, too :+1: 
Code is good from my side :+1:

**Comment by motokito** [comment]
Screenshots are added :smile:

**Comment by koppor** [comment]
@motokito It would have been good if you showed old and new behavior, not just the new one :innocent: 

I have two small remarks, please fix them - otherwise:

[LGTM](http://lgtm.in/i/OT5aIbB6q)

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java [hunk 4]
```diff
@@ -1503,15 +1510,15 @@ public void output(final String s) {
...
         openDatabaseOnlyActions.clear();
         openDatabaseOnlyActions.addAll(Arrays.asList(manageSelectors, mergeDatabaseAction, newSubDatabaseAction, save,
-                saveAs, saveSelectedAs, saveSelectedAsPlain, undo, redo, cut, deleteEntry, copy, paste, mark, unmark,
-                unmarkAll, editEntry, selectAll, copyKey, copyCiteKey, copyKeyAndTitle, editPreamble, editStrings,
-                toggleGroups, makeKeyAction, normalSearch, mergeEntries, cleanupEntries, exportToClipboard, replaceAll,
-                sendAsEmail, downloadFullText, writeXmpAction, findUnlinkedFiles, addToGroup, removeFromGroup,
+                saveAs, saveSelectedAs, saveSelectedAsPlain, editModeAction, undo, redo, cut, deleteEntry, copy, paste, mark, markSpecific, unmark,
+                unmarkAll, rankSubMenue, editEntry, selectAll, copyKey, copyCiteKey, copyKeyAndTitle, editPreamble, editStrings,
+                toggleGroups, makeKeyAction, normalSearch, generalFetcher.getAction(), mergeEntries, cleanupEntries, exportToClipboard, replaceAll,
+                sendAsEmail, downloadFullText, writeXmpAction,optMenueItem, findUnlinkedFiles, addToGroup, removeFromGroup,
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: space missing before `optMenueItem` and rename it to `optMenuItem` - menu is spelled without final `e` in English

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java [hunk 5]
```diff
@@ -1689,6 +1702,20 @@ public void createDisabledIconsForMenuEntries(MenuElement menuElement) {
...
+    public void createDisabledIconsForButtons(Container container) {
+        for (int index = 0; index < container.getComponentCount(); index++) {
+            Component component = container.getComponent(index);
+            if (component instanceof JButton) {
+                JButton button = (JButton) component;
+                if (button.getIcon() instanceof IconTheme.FontBasedIcon) {
+                    button.setDisabledIcon(((IconTheme.FontBasedIcon) button.getIcon()).createDisabledIcon());
+                }
+            } else if (component instanceof JPanel) {
+                createDisabledIconsForButtons((JPanel)component);
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: Space missing before `component`
